pkg/controller/server_status_request_controller.go

```diff
@@ -17,104 +17,100 @@ limitations under the License.
+	statusRequestResyncPeriod = 5 * time.Minute
+)
 
-	client         velerov1client.ServerStatusRequestsGetter
-	lister         velerov1listers.ServerStatusRequestLister
-	pluginRegistry clientmgmt.Registry
-	clock          clock.Clock
+type PluginLister interface {
+	// List returns all PluginIdentifiers for kind.
+	List(kind framework.PluginKind) []framework.PluginIdentifier
 }
 
-func NewServerStatusRequestController(
-	logger logrus.FieldLogger,
-	client velerov1client.ServerStatusRequestsGetter,
-	informer velerov1informers.ServerStatusRequestInformer,
-	pluginRegistry clientmgmt.Registry,
-) *statusRequestController {
-	c := &statusRequestController{
-		genericController: newGenericController("serverstatusrequest", logger),
-		client:            client,
-		lister:            informer.Lister(),
-		pluginRegistry:    pluginRegistry,
-
-		clock: clock.RealClock{},
-	}
-
-	c.syncHandler = c.processItem
-	c.resyncFunc = c.enqueueAllItems
-	c.resyncPeriod = statusRequestResyncPeriod
-
-	informer.Informer().AddEventHandler(
-		cache.ResourceEventHandlerFuncs{
-			AddFunc: func(obj interface{}) {
-				req := obj.(*velerov1api.ServerStatusRequest)
-				key := kubeutil.NamespaceAndName(req)
-
-				c.logger.WithFields(logrus.Fields{
-					"serverStatusRequest": key,
-					"phase":               req.Status.Phase,
-				}).Debug("Enqueueing server status request")
-
-				c.queue.Add(key)
-			},
-		},
-	)
+// ServerStatusRequestReconciler reconciles a ServerStatusRequest object
+type ServerStatusRequestReconciler struct {
+	Scheme       *runtime.Scheme
+	Client       client.Client
+	Ctx          context.Context
+	ServerStatus velero.ServerStatus
 
-	return c
+	Log logrus.FieldLogger
 }
 
-func (c *statusRequestController) processItem(key string) error {
-	log := c.logger.WithField("key", key)
-
-	log.Debug("Running processItem")
-	ns, name, err := cache.SplitMetaNamespaceKey(key)
-	if err != nil {
-		return errors.Wrap(err, "error splitting queue key")
-	}
+// +kubebuilder:rbac:groups=velero.io,resources=serverstatusrequests,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=velero.io,resources=serverstatusrequests/status,verbs=get;update;patch
+func (r *ServerStatusRequestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithFields(logrus.Fields{
+		"controller":          "serverstatusrequest",
+		"serverStatusRequest": req.NamespacedName,
+	})
 
+	// Fetch the ServerStatusRequest instance.
 	log.Debug("Getting ServerStatusRequest")
-	req, err := c.lister.ServerStatusRequests(ns).Get(name)
-	// server status request no longer exists
-	if apierrors.IsNotFound(err) {
-		log.WithError(err).Debug("ServerStatusRequest not found")
-		return nil
+	statusRequest := &velerov1api.ServerStatusRequest{}
+	if err := r.Client.Get(r.Ctx, req.NamespacedName, statusRequest); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.WithError(err).Debug("ServerStatusRequest not found")
```

Comment:
I think this message should be an `Error`, not a `Debug`, as regular users will be using it.